### PR TITLE
Unit Restrictions Improvements

### PR DESCRIPTION
**Pull request by HUSSAR-mtrela**
This PR provides improvements to `UnitsRestrictions.lua` and `UnitsManagers.lua` 
- [x] added icons for new presets and replaced old icons
- [x] added new restriction presets
- [x] removed unused restriction presets
- [x] simplified creation of restriction presets
- [x] changed order of presets in Unit Manager

Also, this PR makes easier for Nomads feature mod to add "No Nomands Units" restriction by overriding `GetPresetsOrder()` function in `UnitsRestrictions.lua` instead of replacing the whole file.

These improvements depend on [PR#1766](https://github.com/FAForever/fa/pull/1766)  which was already merged to fafdevelop branch.

**Units Manager before improvements**
unitsmanager-before
**Units Manager after improvements**
unitsmanager-after

### CODE REVIEW

**Comment by HUSSAR-mtrela** [comment]
this PR is ready for review and I'm open for suggestions/changes

File: lua/ui/lobby/UnitsRestrictions.lua [hunk 1]
```diff
@@ -647,5 +804,6 @@ end
 function GetPresetsOrder()
     -- ensure restriction presets are generated before accessing order 
     GetPresetsData()
+    -- table.insert(presetsOrder, "NOMADS") -- uncomment for NOMADS support
```
Comment (aeoncleanse) [inline lua/ui/lobby/UnitsRestrictions.lua]: Nomads isn't official. Please remove commented code and notify Nomads guys about how to do this.

File: lua/ui/lobby/UnitsRestrictions.lua [hunk 2]
```diff
@@ -647,5 +804,6 @@ end
 function GetPresetsOrder()
     -- ensure restriction presets are generated before accessing order 
     GetPresetsData()
+    -- table.insert(presetsOrder, "NOMADS") -- uncomment for NOMADS support 
     return presetsOrder
 end
```
Comment (aeoncleanse) [inline lua/ui/lobby/UnitsRestrictions.lua]: Files in our codebase require EOL. Please add :)

File: lua/ui/lobby/UnitsManager.lua [hunk 3]
```diff
@@ -362,25 +362,32 @@ end
...
+    GUI.presetsGrid = CreateGrid(GUI.content, cellSize, cellSize, cellMax, rowMax)
     GUI.presetsGrid.Top:Set(function() return GUI.content.Top() + 6 end)
     GUI.presetsGrid.Left:Set(function() return GUI.content.Left() + gridMargin end)
-    GUI.presetsGrid.Height:Set(function() return cellSize * 2 end)
+    GUI.presetsGrid.Height:Set(function() return cellSize * rowMax end)
     GUI.presetsGrid.Width:Set(function() return GUI.content.Width() - gridMargin end)
 
     local index = 0
-    local column = 1
+    local col = 1
```
Comment (aeoncleanse) [inline lua/ui/lobby/UnitsManager.lua]: This rename reduces clarity and readability. Unnecessary.

**Comment by HUSSAR-mtrela** [comment]
@aeoncleanse I made corrections base on your request

**Pull request merged**